workflow/controller/agent.go

```diff
@@ -135,7 +135,7 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 				RunAsUser:    pointer.Int64Ptr(8737),
 			},
 			ServiceAccountName:           serviceAccountName,
-			AutomountServiceAccountToken: pointer.BoolPtr(false),
+			AutomountServiceAccountToken: pointer.BoolPtr(*woc.execWf.Spec.AutomountServiceAccountToken),
```

Comment:
Why de-reference and re-reference? Simpler:

```suggestion
			AutomountServiceAccountToken: woc.execWf.Spec.AutomountServiceAccountToken,
```

Reply:
I think this could change the default behaviour to mount the token. We don’t want that.

Reply:
do you mean that the sidecar(plugin)is designed to that cant mount sa by default?  why ？

Reply:
Correct. Security.

Reply:
one more questuon,how can i init k8s api in sidecar?  Or as I mentioned in [here](https://github.com/argoproj/argo-workflows/issues/7860)
is there a design to make the plugin run forever such as using deployment?